### DISCUSSION TITLE
Yaw Control Subroutine

### DISCUSSION THREAD

**Discussion by yisehakabreham** [NatLabRockies/ROSCO#126] [General]
@rafmudaf,

Does ROSCO activate Yaw Control in its latest release? I would like to test some DLCs on a new floater platform but it seems ROSCO Toolbox does not tune for Yaw rate and gains.

Thanks,
Yisehak A.

**Comment by rafmudaf** [comment]
Yes, ROSCO does have a yaw controller. The current `main` branch has an older version that hasn’t been touched in a very long time, but there is another branch with a more modern controller and this will probably be merged into `develop` in the next few of weeks. There isn’t much tuning that goes into the yaw controller other than a choice of error threshold, so this would probably not be integrated into the ROSCO toolbox.

**Comment by yisehakabreham** [comment]
Thank you @rafmudaf  for clearing up.

I got confused when I find the Yaw Control inputs being zero on ROSCO v2.4.1 Test cases. 
But, it is included in the v2.5.0 release.

**Comment by yisehakabreham** [comment]
Dear Gustavo Hylander,

Thank you for your comment. But, I used the ROSCO Yaw control subroutine
with recommended parameter inputs from v2.5.0 and it works fine on floating
offshore turbines.
I attach below a response of the Nacelle Yaw rotation for a wind direction
change at 300 seconds of the simulation.

Kind regards,
Yisehak A.

[image: image.png]

**Reply by ghylander** [reply]
sorry you are right, i overlooked certain aspects of your use case

**Comment by Dwivedi07** [comment]
Hi @yisehakabreham ,
I am trying to implement my own yaw control algorithm in yaw control subroutine. But when i pass my calculated required yaw rate to avrSWAP(48) the results are not correct. The yaw is not changing as it fluctuates in order of 10^(-3), I tried to test the yaw controller by manually giving constant yaw rate of 2 rad/sec but still I saw the same behaviour. I have turned on Y_controlmode to 1 and changed my Y_ErrThresh(2) to 1 degrees. Can you please help me with this as I have no clue where things are going wrong?
Thanks and regards

**Comment by yisehakabreham** [comment]
@Dwivedi07 ,

Make sure the YCMode in ServoDyn is flagged to Option 5, if you are implementing your yaw control in the DLL. And you should check the YawDOF flag to be True in ElastoDyn.
I don't see other solutions for your case.

Good luck

**Reply by Dwivedi07** [reply]
Thanks for your help!
It worked.
